packages/design-system/src/components/Popover/Popover.tsx

```diff
@@ -13,24 +15,36 @@ import style from './Popover.module.scss';
 
 const ANIMATION_DURATION = 150; // Sync with @talend/design-token animations duration
 
+export type PopoverChildren = ReactNode | ((popover: PopoverStateReturn) => ReactNode);
+
 export type PopoverPropsType = HTMLAttributes<HTMLDivElement> & {
 	disclosure: ReactElement<typeof Clickable>;
-	children: ReactNode | ReactNode[];
+	children: PopoverChildren | PopoverChildren[];
 	position?: Placement;
 } & DataAttributes;
 
 function Popover({ disclosure, position = 'auto', ...props }: PopoverPropsType) {
 	const popover = usePopoverState({ animated: ANIMATION_DURATION, placement: position });
+	const children = Array.isArray(props.children) ? props.children : [props.children];
 
 	return (
 		<>
 			<ReakitPopoverDisclosure {...popover}>
-				{disclosureProps => React.cloneElement(disclosure, disclosureProps)}
+				{disclosureProps => cloneElement(disclosure, disclosureProps)}
 			</ReakitPopoverDisclosure>
-			<ReakitPopover {...popover} {...props}>
+			<ReakitPopover
+				{...popover}
+				style={{ zIndex: tokens.coralElevationLayerStandardFront, ...props.style }}
```

Comment:
`style` is not going to remain as an acceptable prop. `zIndex` would be a standalone optional props.